web/pgadmin/browser/server_groups/servers/databases/publications/tests/test_publication_put.py

```diff
@@ -86,16 +106,181 @@ def runTest(self):
         publication_name = publication_utils. \
             verify_publication(self.server,
                                self.db_name,
-                               self.publication_name)
+                               self.test_data['name'])
+        if not publication_name:
+            raise Exception("Could not find the publication to update.")
+
         if hasattr(self, "update_name"):
             self.test_data['name'] = "test_publication_update_%s" % (
                 str(uuid.uuid4())[1:8])
-        else:
-            self.test_data['name'] = self.publication_name
+
         self.test_data['id'] = self.publication_id
 
-        if not publication_name:
-            raise Exception("Could not find the publication to update.")
+        if hasattr(self, 'add_table'):
+            self.table_name = "table_column_%s" % (str(uuid.uuid4())[1:8])
+            self.table_id = tables_utils.create_table(self.server,
+                                                      self.db_name,
+                                                      self.schema_name,
+                                                      self.table_name)
+            if self.server_version >= 150000:
+                self.test_data['pubtable'] = {'added': [
+                    {'table_name': self.schema_name + '.' + self.table_name}]}
+            else:
+                self.test_data['pubtable'] = publication_utils.get_tables(self)
+
+        if hasattr(self, 'drop_tables'):
+            if self.server_version >= 150000:
+                self.test_data['pubtable'] = {'deleted': [
+                    {'table_name': self.schema_name + '.' + self.table_name}]}
+            else:
+                tables = publication_utils.get_tables(self)
+                self.test_data['pubtable'] = [tables[0]]
+
+        if self.server_version >= 150000:
+            if hasattr(self, 'add_schema'):
+                self.test_data['pubtable'] = ''
+                self.test_data['pubschema'] = \
+                    publication_utils.get_schemas(self)
+            if hasattr(self, 'add_table_with_where'):
```

Comment:
Don't put conditions this way. Use scenarios instead of so many if conditions. Code should behave based on schenarios.